Can't build jjd on OmniOS

Environment:

```
> uname -smr
SunOS 5.11 i86pc
```

Building:

```
[~]> cd jj
[~/jj]> make
CC jjd.c
cc -c -pedantic -Wall -I. -I/usr/include -D_POSIX_C_SOURCE=200809L jjd.c
jjd.c: In function 'input_from_socket':
jjd.c:113:6: warning: implicit declaration of function 'dprintf'; did you mean 'vprintf'? [-Wimplicit-function-declaration]
  113 |  if (dprintf(*pipe_fd, "i %ju %s\n", (uintmax_t)time(NULL), buf) < 0)
      |      ^~~~~~~
      |      vprintf
CC -o jjd
cc -o jjd jjd.o -s -L/usr/lib -lc
Undefined                       first referenced
 symbol                             in file
__xnet_connect                      jjd.o
__xnet_socket                       jjd.o
__xnet_getaddrinfo                  jjd.o
dprintf                             jjd.o
freeaddrinfo                        jjd.o
ld: fatal: symbol referencing errors. No output written to jjd
collect2: error: ld returned 1 exit status
*** Error code 1
make: Fatal error: Command failed for target `jjd'
```

I would imagine that there are more dependencies needed than just a C compiler and awk.

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 PREFIX = /usr
 INCS = -I. -I/usr/include
-LIBS = -L/usr/lib -lc
+LIBS = -L/usr/lib -lc -lsocket
 
 CC = cc
 CPPFLAGS = -D_POSIX_C_SOURCE=200809L

--- a/jjd.c
+++ b/jjd.c
@@ -28,6 +28,27 @@
 static const char *prog;
 static const int *pipe_fd;
 
+#if defined(__sun) && defined(__SVR4)
+int
+dprintf(int fd, const char *restrict format, ...)
+{
+	// As of January 2022, illumos does not have dprintf
+	// yet (see man printf). Provide a wrapper:
+        va_list ap;
+        FILE *f = fdopen(fd, "w");
+
+        if (!f) {
+                return -1;
+        }
+
+        va_start(ap, format);
+        int result = fprintf(f, format, ap);
+        va_end(ap);
+
+        return result;
+}
+#endif
+
 static void
 die(const char *fmt, ...)
 {

